datafusion/physical-expr/src/planner.rs

```diff
@@ -43,14 +43,13 @@ use std::sync::Arc;
 /// * `e` - The logical expression
 /// * `input_dfschema` - The DataFusion schema for the input, used to resolve `Column` references
 ///                      to qualified or unqualified fields by name.
-/// * `input_schema` - The Arrow schema for the input, used for determining expression data types
-///                    when performing type coercion.
 pub fn create_physical_expr(
     e: &Expr,
     input_dfschema: &DFSchema,
-    input_schema: &Schema,
     execution_props: &ExecutionProps,
 ) -> Result<Arc<dyn PhysicalExpr>> {
+    let input_schema: &Schema = &input_dfschema.into();
+
     if input_schema.fields.len() != input_dfschema.fields().len() {
```

Comment:
this condition probably doesn't make sense any longer

Reply:
I agree we can remove it